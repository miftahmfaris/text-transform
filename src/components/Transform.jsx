import React from "react";
import { useSelector } from "react-redux";
import { Grid, Container } from "@material-ui/core";

export default function Transform() {
    const transform = useSelector((state) => state);

    return (
        <Container>
            {transform.upperCase !== undefined && (
                <React.Fragment>
                    <Grid container>
                        <Grid item>
                            <h3>UpperCase : {transform.upperCase}</h3>
                        </Grid>
                    </Grid>
                    <Grid>
                        <Grid item>
                            <h3>AlternateCase : {transform.alternateCase}</h3>
                        </Grid>
                    </Grid>
                </React.Fragment>
            )}
        </Container>
    );
}
