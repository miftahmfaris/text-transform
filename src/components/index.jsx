export { default as FormCharacter } from "./Forms/Character";
export { default as AlternateCase } from "./Transform/AlternateCase";
export { default as UpperCase } from "./Transform/UpperCase";
