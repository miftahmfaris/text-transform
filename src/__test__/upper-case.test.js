import React from "react";
import { render, screen } from "@testing-library/react";
import { UpperCase } from "../components";
import { useSelector } from "react-redux";

jest.mock("react-redux", () => ({
    useDispatch: jest.fn(),
    useSelector: jest.fn(),
}));

describe("<UpperCase />", () => {
    it("should render without error", async () => {
        useSelector.mockImplementation((selector) =>
            selector({
                upperCase: "makan",
                alternateCase: "mAkAn",
            })
        );
        const { container } = render(<UpperCase />);

        expect(container).toMatchSnapshot();
    });
});
