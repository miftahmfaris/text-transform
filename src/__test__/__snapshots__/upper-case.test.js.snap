// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UpperCase /> should render without error 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container"
  >
    <div
      class="MuiGrid-root MuiGrid-item"
    >
      <h3>
        Upper Case : 
        makan
      </h3>
    </div>
  </div>
</div>
`;
